rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java

```diff
@@ -3291,23 +3303,37 @@ private boolean isXmlWorkflow(InputPart fileInputPart) {
 
     @Override
     public RestPage<TaskStateData> getTaskStates(String sessionId, long from, long to, boolean mytasks,
-            boolean running, boolean pending, boolean finished, int offset, int limit)
-                    throws NotConnectedRestException, PermissionRestException {
-        return getTaskStatesByTag(sessionId, null, from, to, mytasks, running, pending, finished, offset, limit);
+                                                 boolean running, boolean pending, boolean finished,
+                                                 int offset, int limit, SortSpecifierRestContainer sortParams)
+            throws NotConnectedRestException, PermissionRestException {
+        String dbSortAttribute = null;
+        SortSpecifierRestContainer filteredSorts = new SortSpecifierRestContainer();
+        if (sortParams != null) {
+            for (SortSpecifierRestContainer.SortSpecifierRestItem i : sortParams.getSortParameters()) {
+                if (sortableTaskAttrMap.containsKey(i.getField())) {
+                    filteredSorts.add(sortableTaskAttrMap.get(i.getField()), i.getOrder());
+                }
+            }
+        }
+        return getTaskStatesByTag(sessionId, null, from, to, mytasks, running, pending, finished, offset, limit,
+                filteredSorts);
     }
 
     @Override
     public RestPage<TaskStateData> getTaskStatesByTag(String sessionId, String taskTag, long from, long to,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 11 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)